### ISSUE TITLE
Fix S2068 Regular Expression creation from parameters

### ISSUE THREAD

**Issue by christophe-zurn-sonarsource** [SonarSource/sonar-dotnet#2783]
If the `CredentialWords` parameter is set to a invald regex expression, such as `PASS)WORD`, rule instanciation will fail and might cause issues for user analysis.
To avoid such an issue, we should make sure that the regex creation does not fail. For instance, by escaping its ivalid characters `this.splitCredentialWords.Select(x => Regex.Escape(x))`. A warning that the parameter is an invalid Regex could also be displayed to the user.
Furthermore, the current regular expression does not catch all cases. Both the following code examples should be noncompliant:
C#
```
foo = "Server=localhost; Database=Main; User=Admin; Password=TopSecret;";       //Noncompliant
foo = "Server=localhost; Database=Main; User=Admin; Password   =   TopSecret;"; //Noncompliant
```
VBNET
```
foo = "Server=localhost; Database=Main; User=Admin; Password=TopSecret;;       'Noncompliant
foo = "Server=localhost; Database=Main; User=Admin; Password   =   TopSecret;; 'Noncompliant
```

New version could be `(?{0})\s*[:=]` (to be validated)

**Comment by pavel-mikula-sonarsource** [comment]
As mentioned in #2768, to avoid FP in 

> "Server=localhost; Database=Main; User=Admin; Password= " + loadPassword()

 the regex should look like `(?{0})\s*[:=]\s*\S`

**Comment by pavel-mikula-sonarsource** [comment]
This was fixed in #3114
